test/unit/mongo_client.test.js

```diff
@@ -110,7 +110,6 @@ describe('MongoOptions', function () {
         return 'very unique';
       }
     },
-    promiseLibrary: global.Promise,
```

Comment:
See other comment, global side effect

Reply:
Since now we expect the provider to have null by default, this is setting the global promise store, so the tests that assert the same promise from the API is ferried along fail